batches_test.go

```diff
@@ -0,0 +1,87 @@
+package packngo
+
+import (
+	"testing"
+	"time"
+)
+
+func TestAccInstanceBatches(t *testing.T) {
+	skipUnlessAcceptanceTestsAllowed(t)
+	c := setup(t)
+
+	c, projectID, teardown := setupWithProject(t)
+	defer teardown()
+
+	req := &InstanceBatchCreateRequest{
+		Batches: []BatchInstance{
+			{
+				Hostname:        "test1",
+				Description:     "test batch",
+				Plan:            "baremetal_0",
+				OperatingSystem: "coreos_stable",
+				Facility:        "ewr1",
+				BillingCycle:    "hourly",
+				Tags:            []string{"abc"},
+				Quantity:        1,
+			},
+		},
+	}
+
+	batches, _, err := c.Batches.Create(projectID, req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var batchID string
+	if len(batches) != 0 {
+		batchID = batches[0].ID
+	}
+
+	batches, _, err = c.Batches.List(projectID, nil)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if batches == nil {
+		t.Fatal("No batches have been created")
+	}
+
+	batch, _, err := c.Batches.Get(batchID, &ListOptions{Includes: "devices"})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var finished bool
+
+	// Wait for all devices to become 'active'
+	for {
+		if len(batch.Devices) == 0 {
+			break
+		}
+		for _, d := range batch.Devices {
+
+			dev, _, _ := c.Devices.Get(d.ID)
+			if dev.State == "active" {
```

Comment:
wondering what's going on here, to me it looks like if any one of the devices in the batch becomes `active`, it will set `finished = true` and not actually wait for all the devices to finish provisioning, which doesn't seem intended?

Reply:
@patrickdevivo good catch, I was testing mostly with a single device in the batch.

Reply:
There's also `waitDeviceActive` defined in `devices_test.go`, could that be utilized as such:

``` golang

for _, d := range batch.Devices {
	waitDeviceActive(d.ID, c)
}
```

The loop should just block until every device is `active` I believe

Reply:
The function is not efficient. It waits until the device with provided ID is "active", the loop I have will block until all devices are "active"

Reply:
Disregard my comment I will have it removed.